NazaDecoder.cpp

```diff
@@ -211,7 +211,8 @@ uint8_t NazaDecoder::decode(int16_t input) {
             if (y < magYMin) {
                 magYMin = y;
             }
-            heading = computeVectorAngle(y - ((magYMax + magYMin) / 2), x - ((magXMax + magXMin) / 2));
+            heading = -atan2(y - ((magYMax + magYMin) / 2), x - ((magXMax + magXMin) / 2)) * 180.0 / M_PI;
```

Comment:
@dalmirdasilva I am not an expert here, but I guess issue is in passing the value. You are passing computeVectorAngle(y, x) and in below function its reversed. [computeVectorAngle(x, y)] x & y are getting interchanged and GPS location is showing 360 on west instead of north.


double Magnetometer::computeVectorAngle(int16_t x, int16_t y) {
    double degrees = radiansToDegrees(-atan2(y, x));
    if (degrees < 0) {
        degrees += 360.0;
    }
    if (degrees > 360.0) {
        degrees -= 360.0;
    }
    return degrees;
}